packages/frontend/src/hooks/useQueryResults.tsx

```diff
@@ -187,21 +187,57 @@ export const useChartResults = (
     invalidateCache?: boolean,
     granularity?: DateGranularity,
 ) => {
+    const isDateZoomFeatureEnabled = useFeatureFlagEnabled('date-zoom');
+    const prevDateZoomGranularityRef = useRef<DateGranularity>();
+    const queryClient = useQueryClient();
+
     const sortKey =
         dashboardSorts
             ?.map((ds) => `${ds.fieldId}.${ds.descending}`)
             ?.join(',') || '';
-    const queryKey = [
-        'savedChartResults',
-        chartUuid,
-        dashboardFilters,
-        invalidateCache,
-        sortKey,
-        granularity,
-    ];
+    const queryKey = useMemo(
+        () => [
+            'savedChartResults',
+            chartUuid,
+            dashboardFilters,
+            invalidateCache,
+            sortKey,
+            granularity,
+        ],
+        [chartUuid, dashboardFilters, invalidateCache, sortKey, granularity],
+    );
+    const apiChartAndResults = queryClient.getQueryData<
+        ApiChartAndResults & { fetched: boolean }
+    >(queryKey);
+
     const timezoneFixFilters =
         dashboardFilters && convertDateDashboardFilters(dashboardFilters);
 
+    useEffect(() => {
+        if (!isDateZoomFeatureEnabled) return;
+
+        // Check if dateZoomGranularity has changed and if date dimensions are present
+        const hasDateZoomChanged =
```

Comment:
this is already present in the queryKey, why do you need to check again here ?

Reply:
we don't! you're right. Thanks for spotting this